x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -204,6 +203,8 @@ export function App({
                   trackUiEvent('app_query_change');
                 }
 
+                data.query.timefilter.timefilter.setTime(dateRange);
```

Comment:
Not sure what effect this has, but is it worth looking at whether the new date range differs from the old date range? There is already a condition above

Reply:
This syncs the time range in the other direction (from within the app to the global state), but you are right, we can put it into that condition.